---
title: "L1000_analysis_using_prot_pair_averages"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Loading data

Load in the L1000_prot_pair_averages.rdata file from github

This is the data created in the L1000_protein_pairs_creating_averages.R script 

We have multiplied the shRNA z-scores by -1

The shRNA is a loss of function perturbagen, the cDNA peturbagen over expresses wild type genes, therefore to make them comparable we multiple the shRNA z-scores by -1

```{r}

githubURL <- "https://github.com/hw15842/L1000_protein_pair_trait_MR/blob/master/Data/L1000_prot_pair_averages.rdata?raw=true"
load(url(githubURL))
head(L1000_prot_pair_averages, n=2)

```

## Investigating the dataset

What are some features of this data

```{r}
summary(L1000_prot_pair_averages$mean)
```

## Extracting significant z-scores 

We use the pnorm() func to turn the z-scores into p-values

We need to use the absolute value of the Z-score with lower.tail set to False to be able to accurately compare the p-values. 
We then also need to multiple the p-value by 2 to get a threshold of 0.05 (rather than 0.25 for a one tailed test)

We determine if it is a significant p-value by dividing 0.05 by 401,080 - this is the number of original protein pairs before adding "forward and reverse" prot pairs 

```{r}

L1000_prot_pair_averages$pval_mean_z <- (pnorm(abs(L1000_prot_pair_averages$mean), lower.tail=F))*2
L1000_prot_pair_averages$sig_zscore_mean <- ifelse(L1000_prot_pair_averages$pval_mean_z < (0.05/401080), "Yes", "No")

table(L1000_prot_pair_averages$sig_zscore_mean)

# Only 1 of the protein pairs have a significant z-score at this threshold

# Just look at 0.05 threshold

L1000_prot_pair_averages$sig_zscore_mean <- ifelse(L1000_prot_pair_averages$pval_mean_z < 0.05, "Yes", "No")

table(L1000_prot_pair_averages$sig_zscore_mean)

```

So there are only 82 protein pairs that are significant at the very high threshold of 0.05


## Looking at maximum rather than average

Rather than using the average of the z-scores, we will look at the largest value. The different z-scores come from the different cell lines in which the analysis took place

We need to determine the **absolute** largest Z-score (i.e. the "largest" negative number or largest positive number)

We can also do FDR correction for multiple testing which is slightly less stringent 


```{r}
L1000_prot_pair_averages$sig_zscore_mean <- NULL

L1000_prot_pair_averages$largest <- ifelse(abs(L1000_prot_pair_averages$min) > abs(L1000_prot_pair_averages$max), L1000_prot_pair_averages$min, L1000_prot_pair_averages$max)

L1000_prot_pair_averages$pval_max_z <- (pnorm(abs(L1000_prot_pair_averages$largest), lower.tail=F))*2

L1000_prot_pair_averages$fdr <- p.adjust(L1000_prot_pair_averages$pval_max_z, "fdr")

table(L1000_prot_pair_averages$fdr < 0.05)

```

This now gives us 194,508 protein pairs that are in the L1000 data with 5,108 protein pairs that pass multiple testing threshold. 




# Comparing L1000 to protein pair MR 

We are going to look at multiple ways of comparing the L1000 protein pairs to our MR results for the protein pairs 


# Can MR predict the directionality of the causal relationship between two proteins?

We are going to look at this in two ways: 

1. Getting the R-squared values and performing stieger filtering to determine directionality then comparing to L1000 direction and FDR values for significance
2. Look at the subset of L1000 data that has fwd and rev prot pairs and compare that to the R-squared values
3. Looking at the beta values to determine directionality

and then we will compare both of these to the directionality as dictated by the L1000 dataset 


## Get the R^2^ values

We need to get the R^2^ values for the SNP on the exposure and the SNP on the outcome for all the protein pairs 

If the SNP-exposure has the larger R^2^ , it might suggest that the exposure influences the outcome and vice versa. 

We can also use stieger filtering to determine if one R^2^ is significantly larger than the other.

We calculated the R^2^ values using the [Calculating_R_squared_prot_pairs.R](https://github.com/hw15842/L1000_protein_pair_trait_MR/blob/master/Scripts/Calculating_R_squared_prot_pairs.R) script which ran the data iteratively to stop it timing out
We combined the R^2^ tables from the iterative analysis in the [combining_rsq_data.R](https://github.com/hw15842/L1000_protein_pair_trait_MR/blob/master/Scripts/combining_rsq_data.R) script 

read in the rsq data table - this is too large to be uploaded via git so have just downloaded it via filezila to my computer


```{r}


load("/Users/hw15842/Documents/PhD_work/Chapter4_L1000_MR_triplicates_analyses/L1000_protein_pair_trait_MR/Data/rsq_dat.rdata")
head(rsq_dat, n=3)
nrow(rsq_dat)
nrow(unique(rsq_dat))

## When creating the rsq_dat file, some of the outcome proteins were in the dataset twice because of there being two seperate pQTLs for them
## We don't need the pQTLs, it just means some have been run twice, so we can now just keep the unique rows

rsq_dat <- unique(rsq_dat)


```

## 1) Does the direction of the R^2^ correlate with the direction of the Z-score? 

Does the direction of the R^2^ match the direction as predicted by the L1000 data for the protein pairs 

**Re-cap**

R^2^

* We have an R^2^ for the exposure and outcome protein in the rsq_dat table 
* the largest R^2^ for the exposure or outcome is the protein that goes "first" 
* steiger filtering can benuse to determine if there is a significant difference between the two R^2^ values 
      
L1000

* the "largest" column is the largest number in either "max" or "min" collumne 
* i.e. the largest absolute Z-score, but has kept the sign (-/+)
* the direction is prot 1 on prot 2 as that is what we extracted from the L1000 dataset
* we also did the reverse so if the reverse was in the dataset it is also in there 
      


```{r}

## keep just the steiger columns from rsq_dat

rsq_dat_steiger_info <- subset(rsq_dat, select=c(id.exposure, id.outcome, SNP, rsq.exposure, rsq.outcome, steiger_dir, steiger_pval))
table(rsq_dat_steiger_info$steiger_dir)

load("/Users/hw15842/Documents/PhD_work/protein_interactions/Data/protein_linker_file.rdata")
head(protein_linker_file, n=2)

```

The protein linker file only has the names of the prot-a proteins,  this is ok as this was also used to create the "all_real_prot_on_prot" data file which was used to extract the data from the L1000 data base. 

```{r}
num_exp_prot_a <- length(grep("prot-a", rsq_dat_steiger_info$id.exposure))
num_exp_prot_a
num_out_prot_a <- length(grep("prot-a", rsq_dat_steiger_info$id.outcome))
num_out_prot_a

nrow(rsq_dat_steiger_info) - num_exp_prot_a
nrow(rsq_dat_steiger_info) - num_out_prot_a
```

so we only loose 2733 expsure proteins and 567 outcome proteins of the total 380,977 protein pairs but just using the prot-a data
This data wasnt extracted from the L1000 data set anyway

```{r}
rsq_dat_steiger_info <- merge(rsq_dat_steiger_info, protein_linker_file, by.x="id.exposure", by.y="protein_number")
names(rsq_dat_steiger_info)[8:9] <- c("Exposure_protein_ID","Exposure_protein_name")

rsq_dat_steiger_info <- merge(rsq_dat_steiger_info, protein_linker_file, by.x="id.outcome", by.y="protein_number")
names(rsq_dat_steiger_info)[10:11] <- c("Outcome_prtein_ID","Outcome_protein_name")

rsq_dat_steiger_info <- rsq_dat_steiger_info[c(2,1,3:11)]  ## put the exposure column first 

head(rsq_dat_steiger_info, n=2)
head(L1000_prot_pair_averages, n=2)
```

For the R^2^ data, the "steiger_dir" column tells us whether the R-squared value is larger for the exposure (TRUE) or the outcome (FALSE).

In the L1000 data we have the fdr adjusted pvalue to tell us if protein 1 significantly influences protein 2 

``` {r}
## Convert the rsq data to be the "correct" direction based in the steiger_dir column

rsq_dat_steiger_info$prot_1 <- ifelse(rsq_dat_steiger_info$steiger_dir == "TRUE", paste0(rsq_dat_steiger_info$Exposure_protein_name), paste0(rsq_dat_steiger_info$Outcome_protein_name))
rsq_dat_steiger_info$prot_2 <- ifelse(rsq_dat_steiger_info$steiger_dir == "TRUE", paste0(rsq_dat_steiger_info$Outcome_protein_name), paste0(rsq_dat_steiger_info$Exposure_protein_name))

head(rsq_dat_steiger_info, n=2)
head(L1000_prot_pair_averages, n=2)

L1000_prot_pair_averages$prot_1 <- as.character(L1000_prot_pair_averages$prot_1)
L1000_prot_pair_averages$prot_2 <- as.character(L1000_prot_pair_averages$prot_2)

rsq_dat_in_L1000 <- subset(rsq_dat_steiger_info, rsq_dat_steiger_info$prot_1 %in% L1000_prot_pair_averages$prot_1 & rsq_dat_steiger_info$prot_2 %in% L1000_prot_pair_averages$prot_2)

head(rsq_dat_in_L1000, n=2)
nrow(rsq_dat_in_L1000)

```

We now want to compare the strength of the associations in the rsq data and the L1000 data so put them all into one table

```{r}

rsq_dat_in_L1000$prot1_on_prot2 <- paste0(rsq_dat_in_L1000$prot_1, "_on_", rsq_dat_in_L1000$prot_2)
L1000_prot_pair_averages$prot1_on_prot2 <- paste0(L1000_prot_pair_averages$prot_1, "_on_", L1000_prot_pair_averages$prot_2)

rsq_and_L1000_dat <- merge(L1000_prot_pair_averages, rsq_dat_in_L1000, by="prot1_on_prot2")
head(rsq_and_L1000_dat, n=2)

```


Now we want to see if the significant steiger pvalue results have a stronger fdr value in the L1000 data
So we can do a T-test to compare the means of the FDR values between the two groupe (sig steiger pval vs non sig steiger pval)

```{r}
sig_steiger_pval <- subset(rsq_and_L1000_dat, rsq_and_L1000_dat$steiger_pval < 0.05)
non_sig_steiger_pval <- subset(rsq_and_L1000_dat, rsq_and_L1000_dat$steiger_pval > 0.05)

t.test(sig_steiger_pval$fdr, non_sig_steiger_pval$fdr)
```

So there is a small significant difference between the means of the two groups, however the 95% confidence intervals are quite small

Does a significant steiger pvalue mean more likely to be a significant fdr value? 

```{r}
(prop.table(table(sig_steiger_pval$fdr<0.05)))*100

(prop.table(table(non_sig_steiger_pval$fdr<0.05)))*100

```

Both just under 3% of the fdr values are signifcant in both the sig and non sig stieger pvalue groups


So it appears as though having a significant stieger p-value does not seem to have much of an influence on whether the L1000 fdr value is significant

## 2) Does the R-squared data predict the same direction when looking at protein pairs that have fwd and rev data in L1000

For the L1000 data that has data for protein 1 on protein 2 and for the reverse direction, which direction has the largest fdr value and does the R-squared data predict the same direction?

To assess this we will perform a binomial test 


* Number of trials = number of L1000 protein pairs that have a fwd and rev
* Probability of success = 0.5 (i.e. 50%)
* Number of times correct = number of times that the steiger direction same as the L1000 direction

```{r}

L1000_fwd_and_rev_pairs <- subset(L1000_prot_pair_averages, L1000_prot_pair_averages$prot_1 %in% L1000_prot_pair_averages$prot_2 & L1000_prot_pair_averages$prot_2 %in% L1000_prot_pair_averages$prot_1)

## Check does the right thing
L1000_fwd_and_rev_pairs_check <- L1000_prot_pair_averages
L1000_fwd_and_rev_pairs_check$prot2_on_prot1 <- paste0(L1000_fwd_and_rev_pairs_check$prot_2, "_on_", L1000_fwd_and_rev_pairs_check$prot_1)
L1000_fwd_and_rev_pairs_check <- subset(L1000_fwd_and_rev_pairs_check, L1000_fwd_and_rev_pairs_check$prot1_on_prot2 %in% L1000_fwd_and_rev_pairs_check$prot2_on_prot1)

nrow(L1000_fwd_and_rev_pairs)
nrow(L1000_fwd_and_rev_pairs_check)
## Same num rows so worked

## now to run the binomial test

L1000_fwd_and_rev_pairs_rsq_dat <- merge(L1000_fwd_and_rev_pairs, rsq_dat_in_L1000, by="prot1_on_prot2")

binom.test(x=nrow(subset(L1000_fwd_and_rev_pairs_rsq_dat, L1000_fwd_and_rev_pairs_rsq_dat$steiger_pval < 0.05)), n=nrow(L1000_fwd_and_rev_pairs_rsq_dat), p=0.5)

```





## 3) Does the *Beta* correlate with the Z-score 

We have the *beta* values for the MR of the protein pairs

Does the direction of the *beta* correlate with the direction of the Z-score 

```{r}



```




# Can MR predict the association of two proteins? (not looking at directionality)

We want to determine if a significant MR result can determine if the protein pair is significant in the L1000 data

We will do this by determining how often a significant MR is significant in the L1000 data and then compare that to some randomly iterated protein pairs

## How often do we get a significant MR and a significant L1000 protein pair 

How often are we finding the a significant MR is also significant in the L1000 data. 

```{r}

```



## does the L1000 hep us to determine if we shold be Removing the mass pleiotropy pQTls

We need to consider and then remove the protein interactions that are due to a mass pleiotropy pQTL as we have determined that these interactions are likely due to biases in the data rather than true biological associations.

We will then also need to change the multiple testing to adjust for this removal of protein pairs 

However I also want to just do some analysis to evaluate if the L1000 data picks up the mass pleiotropy protein pairs, if not adds further credibility to our theory that they are not true biological associations. 

For this section we will use data that has been created above, then after we have removed the mass pleiotropy pQTL associations we will re-run the above analysis in a different order that makes more sense for the flow of the paper 

```{r}

mass_pleio_snps <- read.table("/Users/hw15842/Documents/PhD_work/protein_interactions/new_results/mass_pleio_snps.txt")

rsq_L1000_mass_pleio_assocs <- subset(rsq_and_L1000_dat, rsq_and_L1000_dat$SNP %in% mass_pleio_snps$V1)

rsq_L1000_non_mass_pleio_assocs <- subset(rsq_and_L1000_dat, !rsq_and_L1000_dat$SNP %in% mass_pleio_snps$V1)

(prop.table(table(rsq_L1000_mass_pleio_assocs$fdr<0.05)))*100

(prop.table(table(rsq_L1000_non_mass_pleio_assocs$fdr<0.05)))*100

```


So the percentages of fdr values less than 0.05 are basically exactly the same in the mass pleiotopy and non-mass pleiotropy subsets of the protein pairs 

Doesnt really support our theory that the mass pleio associations are less likely to be biological..... May need to think about this again but for now going to continue with the analysis by removing the mass pleiotropy pQTL associations 





# Removing all of the previous data and re starting with no mass pleio snps and running in more coherent order 

```{r}
rm(list = ls())

ls()
```

/

//

///

////

/////

/////

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

/////

////

///

//

/


#Starting again - no mass pleio pQTLs

## Three questions to answer 

1. Can the MR results predict the same protein pairs as the L1000 data 
2. Can the MR results predict the same protein orientation as the L1000 data 
3. Can the MR results predict the same direction of effect as the L1000 data 


First we need to get the data and remove the mass pleio pQTLs (doing this again as may remove the data from above)

```{r}
# Load in the L1000 results
githubURL <- "https://github.com/hw15842/L1000_protein_pair_trait_MR/blob/master/Data/L1000_prot_pair_averages.rdata?raw=true"
load(url(githubURL))

# determine absolute largest z-score, create a pval and fdr corrected pval for it
L1000_prot_pair_averages$largest <- ifelse(abs(L1000_prot_pair_averages$min) > abs(L1000_prot_pair_averages$max), L1000_prot_pair_averages$min, L1000_prot_pair_averages$max)
L1000_prot_pair_averages$pval_max_z <- (pnorm(abs(L1000_prot_pair_averages$largest), lower.tail=F))*2
L1000_prot_pair_averages$fdr <- p.adjust(L1000_prot_pair_averages$pval_max_z, "fdr")

# load in the rsq_dat
load("/Users/hw15842/Documents/PhD_work/Chapter4_L1000_MR_triplicates_analyses/L1000_protein_pair_trait_MR/Data/rsq_dat.rdata")
## When creating the rsq_dat file, some of the outcome proteins were in the dataset twice because of there being two seperate pQTLs for them
## We don't need the pQTLs, it just means some have been run twice, so we can now just keep the unique rows
rsq_dat <- unique(rsq_dat)

# load in the mass pleiotropy pQTL file 
mass_pleio_snps <- read.table("/Users/hw15842/Documents/PhD_work/protein_interactions/new_results/mass_pleio_snps.txt")


# remove the mass pleiotropy pQTLs from the rsq_dat file 
rsq_dat <- subset(rsq_dat, ! rsq_dat$SNP %in% mass_pleio_snps$V1)


# create data frame with jsut the steiger info 
rsq_dat_steiger_info <- subset(rsq_dat, select=c(id.exposure, id.outcome, SNP, rsq.exposure, rsq.outcome, steiger_dir, steiger_pval, beta.exposure, beta.outcome, se.exposure, se.outcome))

# load in the protein linker file for the prot-a protein names 
load("/Users/hw15842/Documents/PhD_work/protein_interactions/Data/protein_linker_file.rdata")

# Add the protein names to the rsq steiger nmp data frame

rsq_dat_steiger_info <- merge(rsq_dat_steiger_info, protein_linker_file, by.x="id.exposure", by.y="protein_number")
names(rsq_dat_steiger_info)[12:13] <- c("Exposure_protein_ID","Exposure_protein_name")

rsq_dat_steiger_info <- merge(rsq_dat_steiger_info, protein_linker_file, by.x="id.outcome", by.y="protein_number")
names(rsq_dat_steiger_info)[14:15] <- c("Outcome_protein_ID","Outcome_protein_name")

rsq_dat_steiger_info <- rsq_dat_steiger_info[c(2,1,3:15)]  ## put the exposure column first 


# We now have the rsq data with the mass pleio pQTLs removed
# and the L1000 data with the pvalue and fdrs added (this still contains protein pairs that may be due to mass pleio pQTLs at the moment, will remove when we merge with rsq data next)



```

## 1) Can the MR results predict the same protein pairs as the L1000 data 

We want to determine if a significant MR result can determine if the protein pair is significant in the L1000 data

We will do this by determining how often a significant MR is significant in the L1000 data and then compare that to some randomly iterated protein pairs

```{r}

```



## 2) Can the MR results predict the same protein orientation as the L1000 data 

We will assess this in two ways 

* Does a smaller steiger pvalue correlate with a smaller FDR value 
* Does the R^2^ orientation match the L1000 orientation (for the L1000 data that has data about both protein orientations)



#### Does a smaller steiger pvalue correlate with a smaller FDR value 

```{r}

## Convert the rsq data to be the "correct" direction based in the steiger_dir column

rsq_dat_steiger_info$prot_1 <- ifelse(rsq_dat_steiger_info$steiger_dir == "TRUE", paste0(rsq_dat_steiger_info$Exposure_protein_name), paste0(rsq_dat_steiger_info$Outcome_protein_name))
rsq_dat_steiger_info$prot_2 <- ifelse(rsq_dat_steiger_info$steiger_dir == "TRUE", paste0(rsq_dat_steiger_info$Outcome_protein_name), paste0(rsq_dat_steiger_info$Exposure_protein_name))

## Also add in the beta values for the new orientation as well

rsq_dat_steiger_info$prot_1_beta <- ifelse(rsq_dat_steiger_info$steiger_dir == "TRUE", paste0(rsq_dat_steiger_info$beta.exposure), paste0(rsq_dat_steiger_info$beta.outcome))
rsq_dat_steiger_info$prot_2_beta <- ifelse(rsq_dat_steiger_info$steiger_dir == "TRUE", paste0(rsq_dat_steiger_info$beta.outcome), paste0(rsq_dat_steiger_info$beta.exposure))

# Make the L1000 prot columns into charactor vectors

L1000_prot_pair_averages$prot_1 <- as.character(L1000_prot_pair_averages$prot_1)
L1000_prot_pair_averages$prot_2 <- as.character(L1000_prot_pair_averages$prot_2)

# Combine the two data frames - there are no mass pleio prot pairs in the rsq data so this next step wont have them in either 

rsq_dat_steiger_info$prot1_on_prot2 <- paste0(rsq_dat_steiger_info$prot_1, "_on_", rsq_dat_steiger_info$prot_2)
L1000_prot_pair_averages$prot1_on_prot2 <- paste0(L1000_prot_pair_averages$prot_1, "_on_", L1000_prot_pair_averages$prot_2)

rsq_and_L1000_dat <- merge(L1000_prot_pair_averages, rsq_dat_steiger_info, by="prot1_on_prot2")


## Re-calculate a new FDR corrected pvalue for this data as we have now removed the mass pleio pairs 

rsq_and_L1000_dat$fdr_new <- p.adjust(rsq_and_L1000_dat$pval_max_z, "fdr")


# split into sig and non-sig steiger pvals and see if sig difference in the new fdr values between the two groups 
sig_steiger_pval <- subset(rsq_and_L1000_dat, rsq_and_L1000_dat$steiger_pval < 0.05)
non_sig_steiger_pval <- subset(rsq_and_L1000_dat, rsq_and_L1000_dat$steiger_pval > 0.05)

t.test(sig_steiger_pval$fdr_new, non_sig_steiger_pval$fdr_new)

```

So there is a small significant difference between the two groups with the sig steiger pvals having a slightly greater chance of getting a lower FDR pval than the non-sig steiger pvals

Have a look at the percentages and create an odds ratio for this data 

Odds ratio: 
a is the number of times both A and B are present,
b is the number of times A is present, but B is absent,
c is the number of times A is absent, but B is present, and
d is the number of times both A and B are negative.

```{r}

# percentages of the sig and nonsig steiger pvals that have a significant FDR with the new fdr value
(prop.table(table(sig_steiger_pval$fdr_new<0.05)))*100
(prop.table(table(non_sig_steiger_pval$fdr_new<0.05)))*100



# Calculate the odds ratio
library(epitools)

a <- as.numeric(table(sig_steiger_pval$fdr_new<0.05)[2]) # sig steiger and sig FDR
b <- as.numeric(table(sig_steiger_pval$fdr_new<0.05)[1]) # sig steiger and non-sig FDR
c <- as.numeric(table(non_sig_steiger_pval$fdr_new<0.05)[2]) # non-sig steiger and sig FDR
d <- as.numeric(table(non_sig_steiger_pval$fdr_new<0.05)[1]) # non-sig steiger and non-sig FDR


odds_ratio <- (a*d)/(b*c)
odds_ratio
oddsratio.wald(matrix(c(a,b,c,d),nrow = 2, ncol = 2))


```

So the odds ratio is 0.68 with a 95% CI of 0.35  to 1.33

Which means that a significant steiger pvalue, decreases your chances of getting a significant FDR value by 0.68% and the 95% CI  crosses the zero threshold 


#### Does the R^2^ orientation match the L1000 orientation

For the L1000 data that has data for protein 1 on protein 2 and for the reverse orientation, which orientation has the smallest fdr value and does the R-squared data predict the same direction?

To assess this we will perform a binomial test 


* Number of trials = number of L1000 protein pairs that have a fwd and rev
* Probability of success = 0.5 (i.e. 50%)
* Number of times correct = number of times that the steiger direction same as the L1000 direction


```{r}
# Remove the duplicated p1 on p2 rows 
# we have some protein names that are the same for different protein numbers 
p1_same_p2 <- subset(rsq_dat_steiger_info, rsq_dat_steiger_info$prot_1 == rsq_dat_steiger_info$prot_2)
nrow(p1_same_p2)
library(ieugwasr)
p1_same_p2_gwasinfo_exp <- gwasinfo(p1_same_p2$id.exposure)
p1_same_p2_gwasinfo_out <- gwasinfo(p1_same_p2$id.outcome)
cbind(p1_same_p2_gwasinfo_exp[,1:2], p1_same_p2_gwasinfo_out[,1:2])
names(p1_same_p2_gwasinfo_both) <- c("id.exp", "trait.exp", "id.out", "trait.out")
p1_same_p2_gwasinfo_both$same_trait_name <- ifelse(p1_same_p2_gwasinfo_both$trait.exp == p1_same_p2_gwasinfo_both$trait.out, "YES", "NO")
table(p1_same_p2_gwasinfo_both$same_trait_name)
# So majority of them have the same actual trait name even though they are slightly different GWAS and the ones that vary are only very slightly different 
# "Coagulation Factor X"" vs "Coagulation factor Xa"
# "Vascular endothelial growth factor A, isoform 121" vs "Vascular endothelial growth factor A"
# "Ck-beta-8-1" vs "C-C motif chemokine 23" - this one is more different  but looking at gene cards https://www.genecards.org/cgi-bin/carddisp.pl?gene=CCL23 they are aliases 


# remove the one with the 

# create two datasets - one with the correct (forward) orientaion (based on the rsq values) and one with the reverse orientation 

fwd_rsq_df <- rsq_dat_steiger_info[!duplicated(rsq_dat_steiger_info[,c("prot1_on_prot2")])]

                     dat[!duplicated(dat[,c('id','id2')]),]

rev_rsq_df <- rsq_dat_steiger_info
rev_rsq_df$prot2_on_prot1 <- paste0(rev_rsq_df$prot_2, "_on_", rev_rsq_df$prot_1)
rev_rsq_df$prot1_on_prot2 <- NULL

fwd_rsq_df$alphabet_direction <- ifelse( fwd_rsq_df$prot_1 < fwd_rsq_df$prot_2, paste0(fwd_rsq_df$prot_1 ,"_on_", fwd_rsq_df$prot_2), paste0(fwd_rsq_df$prot_2 ,"_on_", fwd_rsq_df$prot_1))

rev_rsq_df$alphabet_direction <- ifelse( rev_rsq_df$prot_1 < rev_rsq_df$prot_2, paste0(rev_rsq_df$prot_1 ,"_on_", rev_rsq_df$prot_2), paste0(rev_rsq_df$prot_2 ,"_on_", rev_rsq_df$prot_1))


# Add in the L1000 data based on the fwd and rev orientations 

fwd_rsq_df <- merge(fwd_rsq_df, L1000_prot_pair_averages, by="prot1_on_prot2")
rev_rsq_df <- merge(rev_rsq_df, L1000_prot_pair_averages, by.x="prot2_on_prot1", by.y = "prot1_on_prot2")

# Just keep the columns we need 
fwd_rsq_df <- subset(fwd_rsq_df, select=c(alphabet_direction, prot1_on_prot2, steiger_pval, fdr))
rev_rsq_df <- subset(rev_rsq_df, select=c(alphabet_direction, prot2_on_prot1, steiger_pval, fdr))

colnames(fwd_rsq_df)[2:4] <- paste("FWD", colnames(fwd_rsq_df)[2:4], sep = "_")
colnames(rev_rsq_df)[2:4] <- paste("REV", colnames(rev_rsq_df)[2:4], sep = "_")


both_fwd_rev <- merge(fwd_rsq_df, rev_rsq_df, by="alphabet_direction")

## now to run the binomial test

# binom.test(x=    , n=nrow(L1000_fwd_and_rev_pairs_rsq_dat), p=0.5)

```







## 3) Can the MR results predict the same direction of effect as the L1000 data 

We have the *beta* values for the MR of the protein pairs - this tells us whether protein 1 up regulates or down regulates protein 2

The Z-scores have a sign which tell us whether protein 1 up reulgates (+) or down regulates (-) protein 2 

(we have already accounted for the two different pertubagen types by multiplying the shRNA z-scores by -1)

Does the direction of the *beta* correlate with the direction of the Z-score 

Does the magnitude of the *beta* correlate with the magnitude of the Z-score 

```{r}
## Calculate the beta of prot 1 on prot 2 

rsq_and_L1000_dat$p1_p2_beta <- as.numeric(rsq_and_L1000_dat$prot_2_beta) / as.numeric(rsq_and_L1000_dat$prot_1_beta)




```